### PR TITLE
Renamer golden test ignores GHC uniques

### PR DESCRIPTION
**Pull request by akrmn**

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/DamlRenamer.hs [hunk 1]
```diff
@@ -77,6 +77,7 @@ runDamlRename damlc damlFile = do
       , "--ghc-option=-dumpdir=" <> tempDir
       , "--ghc-option=-ddump-file-prefix=o."
       , "--ghc-option=-ddump-rn"
+      , "--ghc-option=-dsuppress-uniques"
```
Comment (akrmn) [inline compiler/damlc/tests/src/DA/Test/DamlRenamer.hs:80]: In https://github.com/digital-asset/daml/pull/16379 I missed that the output with `-dumpdir` includes uniques while the previous approach that extracted the renamer output from stderr didn't have them. Thankfully I found this flag which gets us closer to the old behavior. This ensures that this test won't need updating if the unique supply changes (e.g. when adding new functions in `daml-prim` or `daml-stdlib`)

Comment (samuel-williams-da) [reply compiler/damlc/tests/src/DA/Test/DamlRenamer.hs:80]: we're now not testing the behaviour of the uniques at all here, is that okay? do we have tests elsewhere that ensure that part is behaving correctly?

Comment (akrmn) [reply compiler/damlc/tests/src/DA/Test/DamlRenamer.hs:80]: to be clear this wasn't testing the uniques prior to #16379 either. I don't think we test that part directly but it's inherited from GHC which does test it

Comment (samuel-williams-da) [reply compiler/damlc/tests/src/DA/Test/DamlRenamer.hs:80]: Ah right okay

**Review by samuel-williams-da — APPROVED** [review]

**Pull request merged**